domain/application/service/status.go

```diff
@@ -305,9 +343,32 @@ func decodeWorkloadStatus(s *application.StatusInfo[application.WorkloadStatusTy
 	}, nil
 }
 
+func decodeUnitAgentWorkloadStatus(
+	agent *application.UnitStatusInfo[application.UnitAgentStatusType],
+	workload *application.UnitStatusInfo[application.WorkloadStatusType],
+	containerStatus *application.StatusInfo[application.CloudContainerStatusType],
+) (*status.StatusInfo, *status.StatusInfo, error) {
```

Comment:
This as well I think should be re-named to mention we're decoding into the display status.

In fact, I'm not convinced this is really a decode operation.

Reply:
This isn't just a decode (to me, decode => no business logic, just a transformation), it's a calculation.

In fact, the `agent` operations here are completely disjoint from `workload` or `container` or `display`.  I think this method is a bad abstraction.

The caller should just call `decodeUnitAgentStatus` and `unitDisplayStatus` themselves

Reply:
Oops method name was cut off. It's `decodeUnitAgentWorkloadStatus`